pulsar/producer_test.go

```diff
@@ -560,6 +560,38 @@ func TestMessageRouter(t *testing.T) {
 	assert.NotNil(t, msg)
 	assert.Equal(t, string(msg.Payload()), "hello")
 }
+func TestMessageSingleRouter(t *testing.T) {
+	// Create topic with 5 partitions
+	topicAdminURL := "admin/v2/persistent/public/default/my-single-partitioned-topic/partitions"
+	err := httpPut(topicAdminURL, 5)
+	defer httpDelete(topicAdminURL)
+	if err != nil {
+		t.Fatal(err)
+	}
+	client, err := NewClient(ClientOptions{
+		URL: serviceURL,
+	})
+
+	assert.Nil(t, err)
+	defer client.Close()
+
+	producer, err := client.CreateProducer(ProducerOptions{
+		Topic:         "my-single-partitioned-topic",
+		MessageRouter: NewSinglePartitionRouter(),
+	})
+
+	assert.Nil(t, err)
+	defer producer.Close()
+
+	ctx := context.Background()
+
+	ID, err := producer.Send(ctx, &ProducerMessage{
+		Payload: []byte("hello"),
+	})
+	assert.Nil(t, err)
+	assert.NotNil(t, ID)
+
```

Comment:
This test also tests the publishing logic. But it doesn't test the single partition router. We may need to add some assertions for it.

Reply:
Need to expose a function `GetSinglePartition() int` that returns the current partition, then only consume messages from that single partition. 

Any other suggestions?

Reply:
We shouldn't expose the method just for testing.
You can refer to this code:
https://github.com/apache/pulsar-client-go/blob/7f91b2bcd798e1dc6ff27b10a557c8ee5440a83a/pulsar/producer_test.go#L494-L509

Create a consumer, and assert that all the messages are consumed from the same partition topic.

Reply:
That's a good suggestion, I have already made the changes.